resources/kcp/charts/kyma-environment-broker/templates/policy.yaml

```diff
@@ -1,25 +1,32 @@
-apiVersion: authentication.istio.io/v1alpha1
-kind: Policy
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
 metadata:
   name: {{ include "kyma-env-broker.fullname" . }}
-  namespace: kcp-system
+  namespace: {{ .Release.Namespace }}
 spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "kyma-env-broker.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
   peers:
   - mtls:
+      allowTls: true
       mode: PERMISSIVE
-  targets:
-  - name: {{ include "kyma-env-broker.fullname" . }}
-  origins:
-  - jwt:
-      issuer: {{ tpl .Values.oidc.issuer $ | quote }}
+---
+apiVersion: security.istio.io/v1beta1
+kind: RequestAuthentication
+metadata:
+  name: {{ include "kyma-env-broker.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "kyma-env-broker.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  jwtRules:
+    - issuer: {{ tpl .Values.oidc.issuer $ | quote }}
       audiences:
       - {{ .Values.oidc.client | quote }}
-      triggerRules:
```

Comment:
Why path filtering removed for JWT verification?

Reply:
In Istio > 1.5 there is no kind `Policy` anymore. Section `jwtRules` are now moved to the `RequestAuthentication`.

According to the first example from [doc](https://istio.io/latest/docs/reference/config/security/request_authentication/) `includedPaths` settings are not necessary to require JWT token.

According to [doc](https://istio.io/latest/docs/reference/config/security/authorization-policy/) about `AuthorizationPolicy` (first example) request is allowed only when has a valid JWT token that meets the conditions

Reply:
The OIDC JWT validation must be converted to `RequestAuthentication`, otherwise if this Policy is left as is without path filtering, it will be applied for OSB oauth2 endpoints, throwing 401 error on our KCP landscapes.